pkg/network/usm/sharedlibraries/watcher.go

```diff
@@ -233,7 +164,12 @@ func (r *soRegistration) unregisterPath(pathID pathIdentifier) bool {
 	return true
 }
 
-func (w *soWatcher) Stop() {
+func (w *Watcher) Stop() {
+	if w == nil {
+		return
+	}
```

Comment:
I think we shouldn't have such code.
If we failed creating a watcher, there is no reason for the caller to treat everything is fine

Reply:
I think that function is never being called

Reply:
I think what we need to do is return an error in the constructor.
having said that I do see value in being defensive and having proper handling of nil values at the _calee_ level to prevent panics and make the _caller_ code simpler.

For example, in the near future (once we add Istio support) the SSL program may or may not run along with the `sharedlibrary.Watcher` (we can have Istio support enabled but shared library tracing support disable). Without proper handling of the nil case, I have to pretty much move them to the caller code like

```
func (o *sslProgram) Stop() {
        if o.watcher == nil {
                o.watcher.Stop()
        }
}
```

Reply:
While that's not a blocker, having that in the code does allow "silent failures"
instead we should be explicit about the errors and make the appropriate adjustments to the code.
A.K.A - we failed to create SOWatcher? Then the code should know that, and shouldn't try to clear it

Reply:
> instead we should be explicit about the errors and make the appropriate adjustments to the code.
I did add an error to `sharedlibrary.NewWatcher`